### PR TITLE
fix: Dependency update for unstructured component

### PR DESCRIPTION
**Pull request by erichare**
This PR makes two changes to the dependencies of the project:

1. It adds the `pdf` extra package for unstructured, for processing PDFs
2. It bumps the minimum version for `huggingface-hub` which is required for Unstructured.

### CODE REVIEW

**Review by ogabrielluiz — APPROVED** [review]
LGTM

**Review by nicoloboschi — CHANGES_REQUESTED** [review]
Could you explain how the new script `nltk_setup` is intended to be used ?

**Comment by erichare** [comment]
> Could you explain how the new script `nltk_setup` is intended to be used ?

Hi @nicoloboschi 

the unstructured component, and specifically the pdf parsing tools, use nltk behind the scenes. But they assume that these three packages, punkt etc, are already available on the host - if they aren’t, the component will error and ask you to download them first.

so the basic idea is just that every time we deploy a release that we are sure those nltk packages are on the machine and accessible for the nltk library - does that make sense? Maybe there’s another way to go about it, I.,e, the dockerfile orchestrates it? But that’s the idea

**Comment by nicoloboschi** [comment]
@erichare could we make it part of the `langflow run` script?
and at the bootstrap it verifies/download the needed files

The current format is not that helpful in fix/troubleshoot the problem

**Comment by erichare** [comment]
> feat: allow drive loader to read a folder recursively #3572

@nicoloboschi when you get a chance, could you look at the latest commit? I'm not sure whether i put it exactly in the most logical place, but the utility function for downloading the resources i put in setup.py, then the run script imports that function and calls it at startup. If the packages are already available, it'll skip downloading them.

File: src/backend/base/langflow/initial_setup/setup.py [hunk 1]
```diff
@@ -615,3 +616,17 @@ def initialize_super_user_if_needed():
...
+
+# Function to download NLTK packages if not already downloaded
+def download_nltk_resources():
+    nltk_resources = {"tokenizers": ["punkt", "punkt_tab"], "taggers": ["averaged_perceptron_tagger"]}
+
+    for category, packages in nltk_resources.items():
+        for package in packages:
+            try:
+                nltk.data.find(f"{category}/{package}")
+                print(f"{package} ({category}) already exists.")
```
Comment (zzzming) [inline src/backend/base/langflow/initial_setup/setup.py:629]: is there a better logging function than just print() ?
I saw logging.info() is used 11  lines above

**Review by zzzming — APPROVED** [review]

**Review by nicoloboschi — APPROVED** [review]
LGTM

**Pull request merged**